utopia-remix/app/components/sharingDialog.tsx

```diff
@@ -59,26 +60,35 @@ function SharingDialog({ project }: { project: ProjectListing | null }) {
   const setSharingProjectId = useProjectsStore((store) => store.setSharingProjectId)
   const accessRequests = useProjectsStore((store) => store.sharingProjectAccessRequests)
 
-  const accessLevel = React.useMemo(() => {
+  const projectAccessLevel = React.useMemo(() => {
     return asAccessLevel(project?.ProjectAccess?.access_level) ?? AccessLevel.PRIVATE
   }, [project])
 
+  const [accessLevel, setAccessLevel] = React.useState<AccessLevel>(projectAccessLevel)
+
   const projectAccessMatchesSelectedCategory = useProjectAccessMatchesSelectedCategory(project)
 
-  const clearSharingProjectId = React.useCallback(() => {
-    if (!projectAccessMatchesSelectedCategory) {
-      setSharingProjectId(null)
-    }
-  }, [setSharingProjectId, projectAccessMatchesSelectedCategory])
+  const changeAccessFetcherCallback = React.useCallback(
+    (data: unknown) => {
+      if (isLikeApiError(data)) {
+        setAccessLevel(projectAccessLevel)
```

Comment:
just to make sure - if the change call succeeds, the `projectAccessLevel` is also updated to the new access level (and doesn't stay in a stale mode)?

Reply:
yeah – if the call succeeds the access level stay put on the one set there, so yeah it does not get stale (and if other components reference the project they receive its fresh data via the loader update)